src/rgw/rgw_admin.cc

```diff
@@ -6425,26 +6425,36 @@ int main(int argc, const char **argv)
     rgw_bucket bucket;
     RGWBucketInfo bucket_info;
     map<string, bufferlist> attrs;
+    bool bucket_exists = true;
     ret = init_bucket(tenant, bucket_name, bucket_id, bucket_info, bucket, &attrs);
     if (ret < 0) {
-      cerr << "ERROR: could not init bucket: " << cpp_strerror(-ret) << std::endl;
-      return -ret;
+      if (yes_i_really_mean_it) {
```

Comment:
can there be any error that does not connote ENOENT?

Reply:
Good question. I started descending the call tree to try to enumerate all potential error codes. But it's deep and splays out. It would take some effort. But that may be moot. If `init_bucket` returns an error, even if it's not `ENOENT`, we can still try to remove the entry from the reshard log. I think that's likely the more general and more correct behavior we're after.

So... perhaps the variable name `bucket_exists` claims too much for what is truly known, and that perhaps a name such as `bucket_initable` is more appropriate? I'd rather do that than check specifically for `-ENOENT` and then users wouldn't be able to use the functionality with other `bucket_init` errors.

Reply:
Generalizing the error message would also be appropriate for that type of fix.

Reply:
as long as the behavior is safe+correct, we're good;  but yes, it might be easier for the next folks with a softer name